scammer-label-propagation/src/preprocessing/get_data.py

```diff
@@ -0,0 +1,321 @@
+                        headers={"X-API-Key": API_key},
+                    )
+                    data[key] = pd.DataFrame(response.json()['data'])
+                    keys_to_pop.append(key)
+                    # active_queries.pop(key)
+                    logger.debug(f"{central_node}:\t{key} Finished")
+                except Exception as e:
+                    logger.error(e)
+                    logger.debug(f"{central_node}:\t{key} failed after successfully running the query. Re-querying")
+                    api_url = f'https://api.allium.so/api/v1/explorer/queries/{queries[key]}/run-async'
+                    response = requests.post(
+                        api_url,
+                        json={"parameters" :{"addresses":list_of_addresses,"tt":max_datetime}},
+                        headers={"X-API-Key": API_key},
+                    )
+                    active_queries[key] = response.json()["run_id"]
+                    total_retries += 1
+        for key in keys_to_pop:
+            active_queries.pop(key)
+        time.sleep(waiting_time)
+    logger.info(f'{central_node}:\tFinished downloading the data')
+    return data
+
+
+def prepare_labels(df) -> pd.DataFrame:
+    """
+    Converts all the label requests from state into a dataframe with the probability of being an attacker/victim 
+    of an scam. 
+    :param df: pd.DataFrame Dataframe with the label requests
+    :return: pd.DataFrame Dataframe with the probability of being an attacker/victim of an scam
+    """
+    attack_words  = ['phish', 'hack', 'attack', 'Attack', 'scam']
+    victim_words = ['Victim', 'victim', 'benign']
+
+    labeled = []
+    df['entity'] = df['entity'].apply(str.lower)
+    for address in df['entity'].unique():
+        temp_label = {'address': address, 'attacker': 0, 'victim': 0}
+        temp_temp_df = df[df['entity'] == address]
+        for i in range(temp_temp_df.shape[0]):
+            temp_series = temp_temp_df.iloc[i]
+            if any([aw in temp_series['label'] for aw in attack_words]):
+                if temp_series['confidence'] > temp_label['attacker']:
+                    temp_label['attacker'] = temp_series['confidence']
+            elif any([vw in temp_series['label'] for vw in victim_words]):
+                if temp_series['confidence'] > temp_label['victim']:
+                    temp_label['victim'] = temp_series['confidence']
+        labeled.append(temp_label)
+    return pd.DataFrame(labeled)
+
+
+def download_labels_graphql(all_nodes_dict, central_node) -> pd.DataFrame:
+    """
+    Downloads the labels of the nodes in all_nodes_dict. It uses the graphql API of Forta.
+    :param all_nodes_dict: dict Dictionary with the nodes to download the labels from
+    :param central_node: str Central node
+    :return: pd.DataFrame Dataframe with the labels"""
+    logger.info(f'{central_node}\tDownloading the automatic labels')
+    forta_api = "https://api.forta.network/graphql"
```

Comment:
does this request work? According to the docs, we'd need to pass in the API key. 
https://docs.forta.network/en/latest/api-keys/

Reply:
For now it's working, my understanding is that it is not required yet, but it will be in the near future